Django 42

frankystone opened the pull request:
This is for the upcoming release of Django 4.2. I used the alpha version for 4.2.

Our code should work, missing are outdated third party apps [django-messages](https://github.com/arneb/django-messages/tree/master/django_messages) and [django-tagging](https://github.com/Fantomas42/django-tagging), so this won't work yet:
* wiki (works without tagging)
* adding new news
* PM messaging

Removed django-pagination and implemented django's own pagination framework

frankystone commented:
@Noordfrees @hessenfarmer @SirVer Anyone against removing tagging from the News?

Imho tagging has no practical usefulness for the News.

SirVer commented:
I am fine with removing Am 10.02.2023 um 08:54 schrieb kaputtnik ***@***.***>:﻿
@Noordfrees @hessenfarmer @SirVer Anyone against removing tagging from the News?
Imho tagging has no practical usefulness for the News.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

blotter commented on README.txt:
```diff
@@ -129,11 +129,20 @@ production environments, only for development/testing.
+This might be useful for testing thread-safety:
+
+ $ cd mainpage
+ $ gunicorn --workers 4 wlwebsite_wsgi:application
```
entweder
`$ gunicorn --workers=4 mainpage.wlwebsite_wsgi:application`
oder
`$ gunicorn -w 4 mainpage.wlwebsite_wsgi:application`

oder?

frankystone replied:
funktioniert beides. Das funktioniert aber nur im Verzeichnis `mainpage`, also `gunicorn --workers 4 wlwebsite_wsgi:application`:

~~~~
gunicorn --workers 4 wlwebsite_wsgi:application
[2023-02-11 13:06:40 +0100] [12102] [INFO] Starting gunicorn 20.1.0
[2023-02-11 13:06:40 +0100] [12102] [INFO] Listening at: http://127.0.0.1:8000 (12102)
[2023-02-11 13:06:40 +0100] [12102] [INFO] Using worker: sync
[2023-02-11 13:06:40 +0100] [12103] [INFO] Booting worker with pid: 12103
[2023-02-11 13:06:41 +0100] [12104] [INFO] Booting worker with pid: 12104
[2023-02-11 13:06:41 +0100] [12105] [INFO] Booting worker with pid: 12105
[2023-02-11 13:06:41 +0100] [12106] [INFO] Booting worker with pid: 12106
~~~~

blotter commented on local_settings.py.sample:
```diff
@@ -36,10 +36,10 @@ DATABASES = {
-NORECAPTCHA_SITE_KEY = 'dummy'
-NORECAPTCHA_SECRET_KEY = 'dummy'
+# Don't use this key in production!
+RECAPTCHA_PUBLIC_KEY = "6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI"
+RECAPTCHA_PRIVATE_KEY = "6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe"
```
das sollte da raus, zumindest die werte und durch neue ersetzt werden

frankystone replied:
Das sind die offiziellen Wert zum testen, siehe https://developers.google.com/recaptcha/docs/faq#id-like-to-run-automated-tests-with-recaptcha.-what-should-i-do

blotter commented:
im allgemeine sollte in den
`[...]/urls.py`
von
`from django.urls import *`
auf
`from django.urls import re_path[, ...]`
umgestellt werden

frankystone commented:
> im allgemeine sollte in den `[...]/urls.py` von `from django.urls import *` auf `from django.urls import re_path[, ...]` umgestellt werden

Auf lange Sich sollte man evtll auf (path)[https://docs.djangoproject.com/en/4.1/topics/http/urls/#example] umstellen. Ich denke das ist ein eigener PR wert.

frankystone commented:
This should be ready now.

Because django-messages is not maintained anymore i have added [my own fork](https://github.com/frankystone/django-messages) to pip_requirements.txt.

django-tagging which is used in our wiki was taken from jazzband but unfortunately it has a [bug shown in a pullrequest](https://github.com/jazzband/django-tagging/pull/2). I have added a workaround for this. On the long run we may should implement an own solution for tagging wiki articles.

I will make a last test by setting up the website from scratch in the next days and probably check the package versions installed by pip if they need updating.

frankystone commented:
After having some trouble yesterday alpha is running now with this branch.

Problems:
- For some reason installing django-messages form my fork does not work ootb. And for some reason pip doesn't like installing it in edit mode `-e`. Solution (activate the venv!):
    1. uninstall django-messages: `pip uninstall django-messages`
    2. try running `pip install -r pip_requirements.txt`. If this won't work comment the corresponding line in `pip_requirements.txt` and install it by hand: `pip install git+https://github.com/frankystone/django-messages.git@8f6605be2e4f4c05817ed9922c0011aaedf18f57#egg=django_messages`

- There is a new security setting which, if not set, doesn't allow to log in or add e.g. posts:
    - Add `CSRF_TRUSTED_ORIGINS = ['https://name_of_site.org']` to `mainpage/local_settings.py`

frankystone commented:
@Noordfrees please proof read the text changes of commit https://github.com/widelands/widelands-website/pull/418/commits/770e58225561c2d7d7dd6f7caf7e3af432e4b315

frankystone commented:
Thanks a lot!
